fighters/ryu/src/status.rs

```diff
@@ -189,3 +223,111 @@ unsafe extern "C" fn ryu_attack_main_loop(fighter: &mut L2CFighterCommon) -> L2C
     }
     0.into()
 }
+
+// FIGHTER_STATUS_KIND_ESCAPE_AIR
+
+#[status_script(agent = "ryu", status = FIGHTER_STATUS_KIND_ESCAPE_AIR, condition = LUA_SCRIPT_STATUS_FUNC_STATUS_PRE)]
+unsafe fn escape_air_pre(fighter: &mut L2CFighterCommon) -> L2CValue {
+    if VarModule::is_flag(fighter.battle_object, vars::common::instance::PERFECT_WAVEDASH) || fighter.handle_waveland(false, false) {
+        VarModule::on_flag(fighter.battle_object, vars::common::status::SHOULD_WAVELAND);
+        GroundModule::attach_ground(fighter.module_accessor, true);
+        fighter.change_status(FIGHTER_STATUS_KIND_LANDING.into(), false.into());
+        return 0.into();
+    }
+    if !fighter.is_prev_status(*FIGHTER_STATUS_KIND_JUMP_SQUAT) {
+        let mut stick_value_x = ControlModule::get_stick_x(fighter.module_accessor);
+        let stick_value_y = ControlModule::get_stick_y(fighter.module_accessor);
+        let lr =  PostureModule::lr(fighter.module_accessor);
+        if (stick_value_x.abs() > 0.66 && stick_value_y == 0.0) {
+            // airdash gfx
+            if (stick_value_x == 0.0) {
+                stick_value_x = 0.001; // to avoid a divide by zero on the next instruction below
+            }
+            let angle = (stick_value_y/stick_value_x).atan();
+            let norm_stick_pos = Vector3f { x: stick_value_x * -1.0 * lr, y: stick_value_y * -1.0, z: 0.0}.normalize();
+            let pos1 = Vector3f { x: norm_stick_pos.x * 4.0, y: 8.0 + norm_stick_pos.y * 8.0, z: 0.};
+            let pos2 = Vector3f { x: norm_stick_pos.x * 9.0, y: 8.0 + norm_stick_pos.y * 12.0, z: 0.};
+            let rot = Vector3f { x:5.0, y:0.0, z: ((stick_value_x.signum() * 90.0) + 180. * angle/3.14159)};
+            let effect_hash = Hash40::new("sys_down_smoke");
+
+            EffectModule::req_on_joint(fighter.module_accessor, effect_hash, Hash40::new("top"),
+            &pos1, &rot, 0.5, &Vector3f{x: 0.0, y: 0.0, z: 0.0}, &Vector3f{x: 0.0, y: 0.0, z: 0.0}, false, 0, 0, 0);
+
+            // transition to custom airdash status
+            let air_dash_status_kind = CustomStatusModule::get_agent_status_kind(fighter.battle_object, statuses::ryu::AIR_DASH);
```

Comment:
Do not call `L2CFighterBase::change_status` in the pre status. Use `StatusModule::set_status_kind_interrupt` using the retrieved status kind and return `1.into()`

Reply:
Justification in updated comments